src/main/java/org/opensrp/web/rest/UploadController.java

```diff
@@ -0,0 +1,361 @@
+
+	private MultimediaService multimediaService;
+
+	private UploadService uploadService;
+
+	private MultimediaRepository multimediaRepository;
+
+	private OpenmrsIDService openmrsIDService;
+
+	private ClientService clientService;
+
+	private EventService eventService;
+
+	@Autowired
+	public void setObjectMapper(ObjectMapper objectMapper) {
+		this.objectMapper = objectMapper;
+	}
+
+	@Autowired
+	public void setMultimediaService(MultimediaService multimediaService) {
+		this.multimediaService = multimediaService;
+	}
+
+	@Autowired
+	public void setUploadService(UploadService uploadService) {
+		this.uploadService = uploadService;
+	}
+
+	@Autowired
+	public void setMultimediaRepository(MultimediaRepository multimediaRepository) {
+		this.multimediaRepository = multimediaRepository;
+	}
+
+	@Autowired
+	public void setOpenmrsIDService(OpenmrsIDService openmrsIDService) {
+		this.openmrsIDService = openmrsIDService;
+	}
+
+	@Autowired
+	public void setClientService(ClientService clientService) {
+		this.clientService = clientService;
+	}
+
+	@Autowired
+	public void setEventService(EventService eventService) {
+		this.eventService = eventService;
+	}
+
+	private List<Map<String, String>> readCSVFile(MultipartFile file) throws IOException {
+		List<Map<String, String>> csvClients = new ArrayList<>();
+		try (Reader reader = new InputStreamReader(file.getInputStream());
+		     CSVParser parser = new CSVParser(reader, CSVFormat.DEFAULT.withHeader())) {
+
+			List<CSVRecord> records = parser.getRecords();
+			for (CSVRecord record : records) {
+				csvClients.add(record.toMap());
+			}
+		}
+		return csvClients;
+	}
+
+	@PostMapping(headers = { "Accept=multipart/form-data" }, produces = {
+			MediaType.APPLICATION_JSON_VALUE })
+	public Map<String, Object> uploadCSV(@RequestParam("event_name") String eventName,
+			@RequestParam("file") MultipartFile file,
+			@RequestParam(value = "team_id", required = false) String teamID,
+			@RequestParam(value = "team_name", required = false) String teamName,
+			@RequestParam(value = "location_id", required = false) String locationID,
+			Authentication authentication
+	) throws IOException,
+			BusinessLogicException {
+		if (hasSpecialCharacters(eventName)) {
+			logger.error(EVENT_NAME_ERROR_MESSAGE);
+			throw new IllegalArgumentException(EVENT_NAME_ERROR_MESSAGE);
+		}
+
+		String entityId = UUID.randomUUID().toString();
+		String providerId = authentication.getName();
+		MultimediaDTO multimediaDTO = new MultimediaDTO(entityId.trim(), providerId, file.getContentType().trim(), null,
+				FILE_CATEGORY);
+
+		List<Map<String, String>> csvClients = readCSVFile(file);
+
+		UploadValidationBean validationBean = uploadService.validateFieldValues(csvClients, eventName, globalID);
+		if (validationBean.getErrors() != null && validationBean.getErrors().size() > 0) {
+			validationBean.setAnalyzedData(null);
+			throw new BusinessLogicException(objectMapper.writeValueAsString(validationBean));
```

Comment:
Maybe this would be called `UploadValidationException` and you should be passing the `validationBean` instead of serialized string